src/mavoscript.js

```diff
@@ -321,7 +321,62 @@ var _ = Mavo.Script = {
 				return ret;
 			},
 			precedence: 3
-		}
+		},
+		"groupby": {
+			symbol: "by",
+			code: (array, key) => {
+				if (Array.isArray(array) && Array.isArray(key))  { 
+					var propName = key[Mavo.as] === true ? key[Mavo.name] : $.value(key[0], Mavo.toNode, "property");
+					var temp = new Mavo.BucketMap({arrays: true});
+					var ret = [];
+					ret[Mavo.groupedBy] = true;
+
+					for (var i = 0; i < array.length; i++) {
+						const k = i < key.length ? Mavo.value(key[i]) : null;
+						temp.set(k, Mavo.value(array[i]));
+					}	
+
+					for (var [value, items] of temp) {
+						var obj = {"$value": value};
+
+						if (propName !== undefined) {
+							obj[propName] = value;
+						}
```

Comment:
You don’t need quotes for $, they are perfectly valid identifier chars in JS. Same applies to `obj["$items"]` below, it can just be `obj.$items`.

We could shorten the 4 lines above into 1 as:

```js
obj.$value = obj[propName || '$value'] = value;
```

or:

```js
var obj = {$value: value, [propName || '$value']: value};
```

I'm in two minds about whether it's worth it, given a small loss of readability. Up to you!